backend/src/apiserver/resource/resource_manager_test.go

```diff
@@ -4020,6 +4103,33 @@ func TestCreateTask(t *testing.T) {
 	assert.Equal(t, expectedTask, storedTask, "The StoredTask return has unexpected value")
 }
 
+func TestGetArtifactSessionInfo(t *testing.T) {
+	_, manager, _, _, _, _ := initWithExperimentAndPipelineAndRun(t)
+
+	// First Artifact has assigned a bucket_session_info
+	artifact1 := &ml_metadata.Artifact{
+		Id:  intPtr(0),
+		Uri: strPtr("s3://test-bucket/pipeline/some-pipeline-id/task/key0"),
+	}
+
+	config1, _, err1 := manager.GetArtifactSessionInfo(context.Background(), artifact1)
+
+	// Second Artifact has assigned a store_session_info
+	artifact2 := &ml_metadata.Artifact{
+		Id:  intPtr(1),
+		Uri: strPtr("s3://test-bucket/pipeline/some-pipeline-id/task/key1"),
+	}
+
+	// Call the function
+	config2, _, err2 := manager.GetArtifactSessionInfo(context.Background(), artifact2)
+
+	// Assert the results
```

Comment:
A good practice is to assert as soon as possible. You can also reuse the variable names.
```
config1, _, err1 := manager.GetArtifactSessionInfo(context.Background(), artifact1)
assert.NoError(t, err1)
assert.NotNil(t, config1)
```

Reply:
@HumairAK @gregsheremeta thoughts? I had another comment to avoid reusing the err variables, but I'm open for any suggestions.